### PR TITLE
Allow 3 cars per train on the Rocket cars

### PR DESCRIPTION
**Pull request by Gymnasiast**
Allow 3 cars per train on the Rocket cars, for feature parity with RCT1.

### CODE REVIEW

**Comment by Gymnasiast** [comment]
I made the change you requested.

**Comment by IntelOrca** [comment]
Hmm, I think it should probably also be extracted to a new method, `PerformRCT1CompatibilityFixes()` - no doubt you will add more.

**Comment by Gymnasiast** [comment]
Ok, I extracted the compatibility fixes to their own method.

File: src/object/RideObject.cpp [hunk 1]
```diff
@@ -129,6 +124,8 @@ void RideObject::ReadLegacy(IReadObjectContext * context, IStream * stream)
     {
         context->LogError(OBJECT_ERROR_INVALID_PROPERTY, "Nausea multiplier too high.");
     }
+
+    this->PerformRCT1CompatibilityFixes();
```
Comment (IntelOrca) [inline src/object/RideObject.cpp]: Remove `this->`, its inconsistent with the rest of the code base.

File: src/object/RideObject.cpp [hunk 2]
```diff
@@ -444,3 +441,11 @@ void RideObject::ReadLegacyVehicle(IReadObjectContext * context, IStream * strea
     vehicle->special_frames = stream->ReadValue<uint8>();
     stream->Seek(4, STREAM_SEEK_CURRENT);
 }
+
+void RideObject::PerformRCT1CompatibilityFixes()
+{
+    if (String::Equals(this->GetIdentifier(), "RCKC    ")) {
```
Comment (IntelOrca) [inline src/object/RideObject.cpp]: Same here as well.

**Comment by Gymnasiast** [comment]
Ok, dropped it.

**Pull request merged**